### PR TITLE
Migrate docs to Flask

### PR DESCRIPTION
**Pull request by bartaz**
Migrates documentation site from static Jekyll site to flask web app

This is a feature branch, so code was already reviewed in relevant PRs.
This PR is meant for final QA before merging this to master.

### QA

- checkout feature branch
- `./run clean` to make sure to get rid of any artifacts from previous builds
- `./run`, wait for server to start
- go to localhost:8101 or [demo](https://vanilla-framework-canonical-web-and-design-pr-2782.run.demo.haus/)
- browse docs website
- check documentation pages
- check if codepen examples are rendering correctly
- check if navigation and side navigation work, if pages are marked selected properly
- go to examples listing page
- check different examples if they work as expected
- check standalone examples page
- check if standalone examples work as expected

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import json
+import os
+
+# Packages
+import flask
+from canonicalwebteam.flask_base.app import FlaskBase
+from canonicalwebteam.templatefinder import TemplateFinder
+
+
+# Rename your project below
```
Comment (tbille) [inline webapp/app.py]: What should I rename? :P

File: webapp/app.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+# Global context settings
+@app.context_processor
+def global_template_context():
+    with open("package.json") as package_json:
+        version = json.load(package_json)["version"]
+
+    return {
+        "version": version,
+        "current_year": datetime.datetime.now().year,
```
Comment (tbille) [inline webapp/app.py]: Flask base already gives it in it's `context_processor`: https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/context.py#L40

File: webapp/app.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            {"path": example_path, "friendly_name": " / ".join(friendly_parts)}
+        )
+
+    return examples
+
+
+# Global context settings
+@app.context_processor
+def global_template_context():
+    with open("package.json") as package_json:
```
Comment (tbille) [inline webapp/app.py]: is it worth opening every time we load a page the file? Maybe it could be done at the load of the application.

Comment (nottrobin) [reply webapp/app.py]: Good idea. Fixing in https://github.com/canonical-web-and-design/vanilla-framework/pull/2807

File: webapp/app.py (hunk 2 above)
Comment (nottrobin) [reply webapp/app.py]: Good point. Fixing in https://github.com/canonical-web-and-design/vanilla-framework/pull/2807

File: webapp/app.py (hunk 1 above)
Comment (nottrobin) [reply webapp/app.py]: Removed in https://github.com/canonical-web-and-design/vanilla-framework/pull/2807

**Pull request merged**